src/app.rs

```diff
@@ -306,6 +332,55 @@ impl<'repo> App<'repo> {
 
     pub fn error(&mut self, error: GitBrowserError) {
         self.active_error = Some(error);
-        self.mode.push(AppMode::Error);
+        self.mode_history.push(AppMode::Error);
+    }
+
+    pub fn view_blob(&mut self) -> Result<bool, GitBrowserError> {
+        self.external_editor = match self.mode() {
+            AppMode::ViewBlob => {
+                if let Some(pager) = &self.blob_pager {
+                    Some(ExternalEditor::new(&pager.blob, &pager.name, &self.editor))
+                } else {
+                    return Ok(false);
+                }
+            }
+            AppMode::BrowseTrees => {
+                if let Some(page) = self.tree_pages.last() {
+                    let (object, name) = match page.select() {
+                        Some(selection) => selection,
+                        None => return Ok(false),
+                    };
+
+                    if !matches!(object.kind(), Some(ObjectType::Blob)) {
+                        return Ok(false);
+                    }
+
+                    let blob = match object.into_blob() {
+                        Ok(blob) => blob,
+                        Err(_) => {
+                            return Err(GitBrowserError::Error(ErrorKind::BlobReferenceError));
+                        }
+                    };
+
+                    Some(ExternalEditor::new(&blob, &name, &self.editor))
+                } else {
+                    return Ok(false);
+                }
+            }
+            _ => {
+                return Ok(false);
+            }
+        };
+        self.mode_history.push(AppMode::ExternalEditor);
+        if let Some(external_editor) = &mut self.external_editor {
+            external_editor.display();
+        }
+        // We need to go back to the previous mode after the blocking editor display
+        self.back();
+        Ok(true)
+    }
```

Comment:
There might be some easier way to do this in general, but can't now think of any other improvements. But, what do you think about reducing nesting by writing this? Like the following (note, I'm not completely sure this will work),

```suggestion
    pub fn view_blob(&mut self) -> Result<bool, GitBrowserError> {
        self.external_editor = match self.mode() {
            AppMode::ViewBlob => {
                if let Some(pager) = &self.blob_pager {
                    Some(ExternalEditor::new(&pager.blob, &pager.name, &self.editor))
                } else {
                    return Ok(false);
                }
            }
            AppMode::BrowseTrees => {
                let page = match self.tree_pages.last() {
                    Some(page) => page,
                    None => return Ok(false),
                };
    
                let (object, name) = match page.select() {
                    Some(selection) => selection,
                    None => return Ok(false),
                };
    
                if !matches!(object.kind(), Some(ObjectType::Blob)) {
                    return Ok(false);
                }
    
                let blob = object.into_blob().map_err(|_| GitBrowserError::Error(ErrorKind::BlobReferenceError))?;
                Some(ExternalEditor::new(&blob, &name, &self.editor))
            }
            _ => return Ok(false)
        };
        self.mode_history.push(AppMode::ExternalEditor);
        if let Some(external_editor) = &mut self.external_editor {
            external_editor.display();
        }
        // We need to go back to the previous mode after the blocking editor display
        self.back();
        Ok(true)
    }
```